react-native/react/native/pinentry.wrapper.desktop.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from '../base-react'
+import ReactDOM from 'react-dom'
+import Pinentry from '../pinentry'
+import ipc from 'ipc'
+
+class PinentryWrapper extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {}
+    ipc.on('pinentryGotProps', props => {
```

Comment:
So using redux could be overkill (likely not for tracker popup) but this thing could have its own redux store and actions/etc that' interact wit the ipc

So normal store. This would dispatch actions instead of directly calling ipc.send(). The actions would do ipc.send/on and emit other async actions. Normal reducers etc